src/Pact/Types/Term.hs

```diff
@@ -253,6 +254,26 @@ instance Semigroup Gas where
 instance Monoid Gas where
   mempty = 0
 
+newtype MilliGas
+  = MilliGas Int64
+  deriving (Eq, Ord, ToJSON, FromJSON, NFData, Generic)
+
+instance Semigroup MilliGas where
+  (MilliGas a) <> (MilliGas b) = MilliGas (a + b)
+
+instance Monoid MilliGas where
+  mempty = MilliGas 0
+
+microsPerGas :: Int64
+microsPerGas = 1000
+
+gasToMilliGas :: Gas -> MilliGas
+gasToMilliGas (Gas n) = MilliGas (n * microsPerGas)
+
+milliGasToGas :: MilliGas -> Gas
+milliGasToGas (MilliGas n) = Gas (n `div` microsPerGas)
+
```

Comment:
I think we're going to accumulate unnecessary thunks on every call to `milliGasToGas` and `gasToMilliGas` due to `div` and `*`. I don't think we'll build up large chains of these, but it seems like it'd be unnecessary churn that we do alot. Something to follow up on when benchmarking I guess.

Reply:
We can run benchmarks to check for regressions.